shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeRecognizerTest.java

```diff
@@ -0,0 +1,136 @@
+        Collection<DataSource> dataSources = Collections.singleton(datasource);
+        assertThat(DatabaseTypeRecognizer.getDatabaseType(dataSources).getName(), is("SQLServer"));
+    }
+
+    @Test
+    public void assertGetDefaultDatabaseType() {
+        assertThat(DatabaseTypeRecognizer.getDatabaseType(Collections.emptyList()).getName(), is("MySQL"));
+    }
+
+    @Test
+    public void assertGetDatabaseTypeFromSameDataSources() throws SQLException {
+        DataSource datasource1 = mockDataSource(DatabaseTypeRegistry.getActualDatabaseType("MySQL"));
+        DataSource datasource2 = mockDataSource(DatabaseTypeRegistry.getActualDatabaseType("MySQL"));
+        Collection<DataSource> sameDataSources = Arrays.asList(datasource1, datasource2);
+        assertThat(DatabaseTypeRecognizer.getDatabaseType(sameDataSources).getName(), is("MySQL"));
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void assertGetDatabaseTypeFromDifferentDataSources() throws SQLException {
+        DataSource datasource1 = mockDataSource(DatabaseTypeRegistry.getActualDatabaseType("H2"));
+        DataSource datasource2 = mockDataSource(DatabaseTypeRegistry.getActualDatabaseType("Oracle"));
+        Collection<DataSource> differentDataSources = Arrays.asList(datasource1, datasource2);
+        DatabaseTypeRecognizer.getDatabaseType(differentDataSources);
+    }
+
+    private DataSource mockDataSource(final DatabaseType databaseType) throws SQLException {
+        DataSource result = mock(DataSource.class);
+        Connection connection = mock(Connection.class, RETURNS_DEEP_STUBS);
+        when(result.getConnection()).thenReturn(connection);
+        if (databaseType instanceof H2DatabaseType) {
+            when(connection.getMetaData().getURL()).thenReturn("jdbc:h2:mem:test;DB_CLOSE_DELAY=-1;DATABASE_TO_UPPER=false;MODE=MySQL");
+        } else if (databaseType instanceof MariaDBDatabaseType) {
+            when(connection.getMetaData().getURL()).thenReturn("jdbc:mariadb://localhost:3306/test");
+        } else if (databaseType instanceof MySQLDatabaseType) {
+            when(connection.getMetaData().getURL()).thenReturn("jdbc:mysql://localhost:3306/test");
+        } else if (databaseType instanceof OracleDatabaseType) {
+            when(connection.getMetaData().getURL()).thenReturn("jdbc:oracle:oci:@127.0.0.1/test");
+        } else if (databaseType instanceof PostgreSQLDatabaseType) {
+            when(connection.getMetaData().getURL()).thenReturn("jdbc:postgresql://localhost:5432/test");
+        } else if (databaseType instanceof SQL92DatabaseType) {
+            when(connection.getMetaData().getURL()).thenReturn("jdbc:sqlserver://127.0.0.1;DatabaseName=test");
+        } else if (databaseType instanceof SQLServerDatabaseType) {
+            when(connection.getMetaData().getURL()).thenReturn("jdbc:microsoft:sqlserver://127.0.0.1;DatabaseName=test");
+        }
+        return result;
+    }
+
```

Comment:
Hi @MartinDai Thanks for this contribution, basically it looks nice, could you remove this redundant line?